spec/view_models/profile_index_spec.rb

```diff
@@ -1,30 +1,188 @@
+    it 'returns personal key item heading partial' do
+      expect(unverified_view_model.personal_key_item_partial).to(
+        eq('profile/personal_key_item_heading')
+      )
+    end
+  end
+
+  describe '#recent_event_partial' do
+    it 'returns partial to format a single recent user event' do
+      expect(unverified_view_model.recent_event_partial).to eq('profile/event_item')
+    end
+  end
+
+  context 'totp related methods' do
+    context 'with totp enabled' do
+      before do
+        user = build_stubbed(:user, otp_secret_key: '123').decorate
+        @view_model = ProfileIndex.new(
+          decrypted_pii: nil,
+          personal_key: nil,
+          decorated_user: user
+        )
+      end
+
+      describe '#totp_partial' do
+        it 'returns a partial to disable totp if active' do
+          expect(@view_model.totp_partial).to eq('profile/actions/disable_totp')
+        end
+      end
+
+      describe '#totp_content' do
+        it 'returns auth app enabled message' do
+          expect(@view_model.totp_content).to eq('profile.index.auth_app_enabled')
+        end
+      end
+    end
+
+    context 'with totp disabled' do
+      describe '#totp_partial' do
+        it 'returns a partial to enable totp' do
+          expect(unverified_view_model.totp_partial).to eq('profile/actions/enable_totp')
+        end
+      end
+
+      describe '#totp_content' do
+        it 'returns auth app disabled message' do
+          expect(unverified_view_model.totp_content).to eq('profile.index.auth_app_disabled')
+        end
+      end
+    end
+  end
+
+  describe '#header_personalization' do
+    it 'returns an email address when user does not have a verified profile' do
+      user = unverified_view_model.decorated_user
+      expect(unverified_view_model.header_personalization).to eq(user.email)
+    end
+
+    it 'returns the users first name when they have a verified profile' do
+      expect(verified_profile_index.header_personalization).to eq('Alex')
+    end
+  end
+
+  describe '#recent_events' do
+    it 'exposes recent_events method from decorated_user' do
+      expect(unverified_profile_index).to respond_to(:recent_events)
+    end
+  end
+end
+
+def verified_profile_index(personal_key: nil)
+  profile = create(:profile, :active, :verified, pii: { first_name: 'Alex' })
+  user = profile.user
+  user_access_key = user.unlock_user_access_key(user.password)
+  decrypted_pii = profile.decrypt_pii(user_access_key)
+
+  ProfileIndex.new(
+    decrypted_pii: decrypted_pii,
+    personal_key: personal_key,
+    decorated_user: user.decorate
+  )
+end
+
+def unverified_profile_index(personal_key: nil)
+  ProfileIndex.new(
+    decrypted_pii: nil,
+    personal_key: personal_key,
+    decorated_user: unverified_decorated_user
+  )
+end
+
+def expect_null_partial(view_model, method)
+  expect(view_model.send(method.to_sym)).to eq('shared/null')
+end
+
+def unverified_decorated_user
+  build_stubbed(:user).decorate
 end
```

Comment:
I don't think so? I put a few helper methods outside of the top-level describe block, should they go inside of it?